mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -441,6 +441,54 @@ def GENX_Matrix2DBlockLoadOp : GENX_Op<"matrix.2Dblockload">,
   let hasVerifier = 1;
 }
 
+def GENX_Matrix2DBlockStoreOp : GENX_Op<"matrix.2Dblockstore">,
+  Arguments<(ins 
+    Arg<LLVM_PointerTo<GENX_MatrixElemType>, "", [MemRead]>:$ptr,
+    I32:$base_width,
+    I32:$base_height,
+    I32:$base_pitch,
+    I32:$x,
+    I32:$y,
+    I32Attr:$elem_size_in_bits,
+    I32Attr:$tile_width,
+    I32Attr:$tile_height,
+    I32Attr:$v_blocks,
+    I1Attr:$transpose,
+    I1Attr:$vnni_transform,
+    AnyInteger:$stored_val
```

Comment:
This looks incorrect. How do we store fp16, bf16, fp32 ?

Reply:
According to https://github.com/intel-innersource/drivers.gpu.unified/blob/comp/igc/Source/IGC/GenISAIntrinsics/Intrinsic_definitions.py#L2283, stored_val should be any int.

Reply:
I think the operation we design needs to store a block of elements so IMO this operand should have type vector< GENX_MatrixElemType>. I should be able to load a block or element and store it back using this operation.

Reply:
The hardware 2D load/store instructions are type-less (just the bits) so IGC intrinsics defines the store value as anyint.  We will need to bitcast the vector of fp16, bf16, fp32, int8 to vector of int32 when we translate this to IGC function call, so we can make the $stored_val to be any FixedVectorOf<[GENX_MatrixElemType]>.

Reply:
I think that the store value vector shape doesn't change:  vector<8xfp32> cast to vector<8xint32>, vector<8xbf16> cast to vector<8xint16>, vector<8xbf16> cast to vector<8xint16>, and etc. should work.

Reply:
@pengtu How can I store a vector of int32 to an integer?

Reply:
@whitneywhtsang: I think that we can pass int32 or int16 type as store the value type to llvm::GenISAIntrinsic::getDeclaration call so IGC will get the right function, and pass the vector store value to the function call.

Reply:
This way, we don't even need to bitcast to int.

Reply:
ok, so `GenISA_LSC2DBlockWrite` accept vector of any element type as store value, and we always pass `int32` to `getDeclaration` to get the  right function?

Reply:
With the latest clarification from IGC, we can pass vector of {int32, int16, int8} to "anyint" based on the original vector value's element type.

Reply:
Thanks for clarifying, what should we do when element type is fp?

Reply:
Given a store value of vector<totalSize x elemType>, we create a vector type of <totalSize/16 x intElemSize_t> as the store value type to getDeclaration call.

Reply:
A further clarification on this SIMT function that applies to a work item (simd lane): each work item supplies its own part of the submatrix values, i.e., one row of the block.  For PVC SIMD 16, if the submatrix block is an array of "float a[16, 4]", the stored value will be a[workitem_id, 0:3] -- a vector<4xf32>.